### PR TITLE
CYS: Add DELETE private/ai/products endpoint

### LINKED ISSUE #12067: Store Customization > Enable DELETE request for the ai/products endpoint
Similarly to what we did on https://github.com/woocommerce/woocommerce-blocks/issues/11873, we need to enable the DELETE request for the `ai/products` route within Store API to ensure all products have content reset to default.

This route should invoke a newly created method called `reset_products_content()` (or similar) within the `ProductUpdater` class. This method should:

1. Fetch the dummy products to update `$dummy_products_to_update = $this->fetch_dummy_products_to_update();`
2. Assign the default content available within the [`DUMMY_PRODUCTS`](https://github.com/woocommerce/woocommerce-blocks/blob/d51816f2253a341ac33874ce497849c259ae9f2a/src/Patterns/ProductUpdater.php#L11) constant to each one of the selected dummy products.

### PR DESCRIPTION
**Pull request by gigitux**
## What

Fixes #12067

## Why

This PR introduces a new endpoint to reset the products for the CYS AI offline flow.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Follow the testing instructions of  https://github.com/woocommerce/woocommerce/pull/41923

## WooCommerce Visibility

Required:

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

### CODE REVIEW

**Review by nefeline — APPROVED** [review]
Great work @gigitux ! Confirmed all dummy products are reset when making a DELETE request to `'/wc/private/ai/products'`:


https://github.com/woocommerce/woocommerce-blocks/assets/15730971/10fcdb0e-a076-4d8a-ab3a-01873f66907c

I've just pushed a change ( 138dc8b94ca5fe00bbd2865b2d138b90b9d9e590 ) to update the hash for the dummy product when the content is reset, otherwise, it wouldn't be updated with AI-generated content on the subsequent successful connections to AI as the `$current_product_hash` would not be equal to `$ai_modified_product_hash` anymore on https://github.com/woocommerce/woocommerce-blocks/blob/138dc8b94ca5fe00bbd2865b2d138b90b9d9e590/src/Patterns/ProductUpdater.php#L182

**Comment by gigitux** [comment]
> Great work @gigitux ! Confirmed all dummy products are reset when making a DELETE request to `'/wc/private/ai/products'`:
> 
>  Screen.Recording.2023-12-06.at.17.48.31.mov 
> I've just pushed a change ( [138dc8b](https://github.com/woocommerce/woocommerce-blocks/commit/138dc8b94ca5fe00bbd2865b2d138b90b9d9e590) ) to update the hash for the dummy product when the content is reset, otherwise, it wouldn't be updated with AI-generated content on the subsequent successful connections to AI as the `$current_product_hash` would not be equal to `$ai_modified_product_hash` anymore on
> 
> https://github.com/woocommerce/woocommerce-blocks/blob/138dc8b94ca5fe00bbd2865b2d138b90b9d9e590/src/Patterns/ProductUpdater.php#L182

Yeah! Good point! Thanks for pushing the commit! I will wait for the E2E tests to run and after that, I will merge the PR! 🙇

**Pull request merged**
